j5/backends/console/sr/v4/motor_board.py

```diff
@@ -0,0 +1,69 @@
+"""Console Backend for the SR v4 Motor Board."""
+
+from typing import List, Optional, Type
+
+from j5.backends import Backend
+from j5.backends.console import Console, ConsoleEnvironment
+from j5.boards import Board
+from j5.boards.sr.v4.motor_board import MotorBoard
+from j5.components.motor import MotorInterface, MotorSpecialState, MotorState
+
+
+class SRV4MotorBoardConsoleBackend(
+    MotorInterface,
+    Backend,
+):
+    """The console implementation of the SR v4 motor board."""
+
+    environment = ConsoleEnvironment
+    board = MotorBoard
+
+    @classmethod
+    def discover(cls) -> List[Board]:
+        """Discover boards that this backend can control."""
+        raise NotImplementedError("The Console Backend cannot discover boards.")
+
+    def __init__(self, serial: str, console_class: Type[Console] = Console) -> None:
+        self._serial = serial
+
+        # Initialise our stored values for the state.
+        self._state: List[MotorState] = [
+            MotorSpecialState.BRAKE
+            for _ in range(0, 2)
+        ]
+
+        # Setup console helper
+        self._console = console_class(f"{self.board.__name__}({self._serial})")
+
+    def get_firmware_version(self) -> Optional[str]:
+        """The firmware version reported by the board."""
+        return None  # Console, so no firmware
+
+    @property
+    def serial(self) -> str:
+        """The serial number reported by the board."""
+        return self._serial
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """The firmware version reported by the board."""
+        return self.get_firmware_version()
+
+    def get_motor_state(self, identifier: int) -> MotorState:
+        """Get the current motor state."""
+        # We are unable to read the state from the motor board, in hardware
+        # so instead of asking, we'll get the last set value.
+        return self._state[identifier]
+
+    def set_motor_state(self, identifier: int, power: MotorState) -> None:
+        """Set the state of a motor."""
+        if identifier not in range(0, 2):
+            raise ValueError(
+                f"Invalid motor identifier: {identifier}, valid values are: 0, 1",
+            )
+        self._state[identifier] = power
+        if isinstance(power, MotorSpecialState):
```

Comment:
`try: except:` ?

Reply:
How would you do that without breaking the typing?

The following doesn't pass `mypy`.

```python
        try:
            power_human_name = power.name
        except AttributeError:
            power_human_name = str(power)
```

Reply:
What's the error?

Reply:
`j5/backends/console/sr/v4/motor_board.py:66: error: Item "float" of "Union[float, MotorSpecialState]" has no attribute "name"`

Reply:
Hmm... this feels like an issue here with `mypy` as it isn't recognising the try / except, except I don't think I would expect it to... leading to is it too much to permit both floats and MotorSpecialState... ? 

As I'm lazy, I think we should just go with look before you leap for this case save further effort from discussion?

Reply:
See https://github.com/python/mypy/issues/2420

Using `if isinstance ...` appears to be the correct approach according to BDFL.